server/src/main/java/org/elasticsearch/cluster/metadata/DataLifecycle.java

```diff
@@ -131,21 +131,14 @@ public int hashCode() {
     public void writeTo(StreamOutput out) throws IOException {
         if (out.getTransportVersion().onOrAfter(TransportVersion.V_8_500_007)) {
             out.writeOptionalWriteable(dataRetention);
-        } else {
-            out.writeOptionalTimeValue(getEffectiveDataRetention());
         }
     }
 
     public DataLifecycle(StreamInput in) throws IOException {
         if (in.getTransportVersion().onOrAfter(TransportVersion.V_8_500_007)) {
             dataRetention = in.readOptionalWriteable(Retention::read);
         } else {
-            var value = in.readOptionalTimeValue();
-            if (value == null) {
-                dataRetention = null;
-            } else {
-                dataRetention = new Retention(value);
-            }
```

Comment:
As `8.8.0` is disabled (except for test clusters) I'm tempted to not add any special handling here

Reply:
Same here? Assuming my understanding is correct I think this makes sense. This is a feature flag, so we do not have to provide guarantees for a cluster moving from `8.8.0` to `8.9.0`, so our main concern is the mixed tests that use the `8.8.1`? Right?